### DISCUSSION TITLE
Performance question regarding many observers

### DISCUSSION THREAD

**Discussion by Zerebokep** [TanStack/query#7473] [Q&A] [answered]
Hey,

I'm currently using useQuery in some of my list items, which results in some scenarios in hundreds of listeners. Is this a good practice or should I switch to reading directly from the cache using queryClient where possible?

**Comment by TkDodo** [comment, marked as answer]
you can't read from the queryClient during rendering. The only other option is passing data in as a prop from a parent.

observers have an overhead but usually negligible. We had reports of some slowness when it goes into the thousands, but hundreds should be totally fine.
